extension/android/CMakeLists.txt

```diff
@@ -21,9 +21,40 @@ include(${EXECUTORCH_ROOT}/build/Utils.cmake)
 set(_common_compile_options -Wno-deprecated-declarations -fPIC)
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
 
-add_subdirectory(
-  ${EXECUTORCH_ROOT}/examples/third-party/fbjni
-  ${CMAKE_CURRENT_BINARY_DIR}/third-party/fbjni
+# We need to download fbjni library from maven, and use its "prefab" library
+# and headers, and link executorch library against that fbjni library.
+# We don't know which NDK is used to compile fbjni, and we need to link our
+# executorch library to the version which Android APK links against for runtime
+# to ensure the libc++ dependencies are consistent.
+# WARNING #
+# Users need to use the SAME fbjni version here and in app gradle dependency
+# for runtime compatibility!
+if(NOT FBJNI_VERSION)
+  set(FBJNI_VERSION 0.5.1)
+endif()
+
+set(FBJNI_AAR_URL https://repo1.maven.org/maven2/com/facebook/fbjni/fbjni/${FBJNI_VERSION}/fbjni-${FBJNI_VERSION}.aar)
+set(FBJNI_DOWNLOAD_PATH ${CMAKE_CURRENT_BINARY_DIR}/third-party/fbjni/fbjni.aar)
```

Comment:
This seems risky? How do you make sure things under `third-party/fbjni` is compatible with `fbjni.aar` downloaded from Maven?

Reply:
Sorry I need to remove examples/third-party/fbjni in a follow up PR. We can't use it. We need to make sure the runtime so deps is the same as compile time so library. We can't build the so from source, because runtime uses another one. The `fbjni.aar` downloaded from Maven basically contains a "SDK" of headers and so library for us to link our ET library against.